tests/async/tasyncawait.nim

```diff
@@ -12,20 +12,21 @@ const
 var clientCount = 0
 
 proc sendMessages(client: AsyncFD) {.async.} =
-  for i in 0 .. <messagesToSend:
+  for i in 0 ..< messagesToSend:
     await send(client, "Message " & $i & "\c\L")
 
 proc launchSwarm(port: Port) {.async.} =
-  for i in 0 .. <swarmSize:
-    var sock = newAsyncNativeSocket()
+  for i in 0 ..< swarmSize:
+    var sock = createAsyncNativeSocket()
 
     await connect(sock, "localhost", port)
     await sendMessages(sock)
     closeSocket(sock)
 
 proc readMessages(client: AsyncFD) {.async.} =
+  var a = newAsyncSocket(client)
```

Comment:
please name this something more descriptive like `sockObj`.

Reply:
Also, a comment might help, took me a few seconds to understand why you were creating a new socket. You're actually wrapping the AsyncFd into a AsyncSocket object.